lifecycle.Rmd

```diff
@@ -7,52 +7,251 @@ status("drafting")
+    This makes in-development versions special and, in some sense, degenerate.
+    Since we don't increment the development component with each Git commit, the same package version number is associated with many different states of the package source, in between releases.
 
 The advice above is inspired in part by [Semantic Versioning](https://semver.org) and by the [X.Org](https://www.x.org/releases/X11R7.7/doc/xorg-docs/Versions.html) versioning schemes.
 Read them if you'd like to understand more about the standards of versioning used by many open source projects.
+But we should underscore that our practices our inspired by these schemes and are somewhat less regimented.
 Finally, know that other maintainers follow different philosophies on how to manage the package version number.
 
-The version number of your package increases with subsequent releases of a package, but it's more than just an incrementing counter -- the way the number changes with each release can convey information about what kind of changes are in the package.
-We discuss this and more in @sec-release-version.
-For now, just remember that the first version of your package should be `0.0.0.9000`.
-`usethis::create_package()` does this, by default.
-`usethis::use_version()` increments the package version; when called interactively, with no argument, it presents a helpful menu:
+## Major vs. minor vs. patch release
+
+The version number of your package is always increasing, but it's more than just an incrementing counter -- the way the number changes with each release can convey information about the nature of the changes.
+The transition from 0.3.1 to 0.3.2, which is a patch release, has a very different vibe from the transition from 0.3.2 to 1.0.0, which is a major release.
+A package version number can also convey information about where the package is in its lifecycle.
+For example, the version 1.0.0 often signals that the public interface of a package is considered stable.
+
+How do you decide which type of release to make, i.e. which component(s) of the version should you increment?
+A key concept is whether the associated changes are backward compatible, meaning that pre-existing code will still "work" with the new version.
+We put "work" in quotes, because this designation is open to a certain amount of interpretation.
+A hardliner might take this to mean "the code works in exactly the same way, in all contexts, for all inputs".
+A more pragmatic interpretation is that "the code still works, but could produce a different result in some edge cases".
+A change that is not backward compatible is often described as a *breaking* change.
+In section @sec-lifecycle-backward-compatibility, we'll revisit the topic of backward compatibility.
+Similar to the issue of package dependencies (@sec-dependencies-pros-cons), it's a question of trade-offs.
+
+In practice, backward compatibility is not a clear-cut distinction.
```

Comment:
Our definition is something like: A breaking changes causes a CRAN packages that previously passed R CMD check to now fail R CMD check AND that the previously behaviour was correct. (i.e. if we make a change to a package that reveals a bug in your package, we don't consider that to be a breaking change).

But maybe we need different terminology for those cases?

Reply:
I worked these specifics in.